src/lib/chips/chip.ts

```diff
@@ -110,14 +142,110 @@ export class MdChip extends _MdChipMixinBase implements Focusable, OnDestroy, Ca
     this.onFocus.emit({chip: this});
   }
 
+  /**
+   * Allows for programmatic removal of the chip. Called by the MdChipList when the DELETE or
+   * BACKSPACE keys are pressed.
+   *
+   * Note: This only informs any listeners of the removal request, it does **not** actually remove
+   * the chip from the DOM.
+   */
+  remove(): void {
+    if (this.removable) {
+      this.onRemove.emit({chip: this});
+    }
+  }
+
+  /** The aria-disabled state for the chip */
+  _isAriaDisabled(): string {
+    return String(this.disabled);
+  }
+
   /** Ensures events fire properly upon click. */
   _handleClick(event: Event) {
     // Check disabled
     if (this.disabled) {
+      return;
+    }
+
+    event.preventDefault();
+    event.stopPropagation();
+
+    this.focus();
+  }
+
+  /** Handle custom key presses. */
+  _handleKeydown(event: KeyboardEvent) {
+    if (this.disabled) {
+      return;
+    }
+
+    switch (event.keyCode) {
+      case DELETE:
+      case BACKSPACE:
+        // If we are removable, remove the focused chip
+        this.remove();
+        // Always prevent so page navigation does not occur
+        event.preventDefault();
+        break;
+      case SPACE:
+        // If we are selectable, toggle the focused chip
+        if (this.selectable) {
+          this.toggleSelected();
+        }
+
+        // Always prevent space from scrolling the page since the list has focus
+        event.preventDefault();
+        break;
+    }
+  }
+
+  protected _checkDisabled(event: Event): boolean {
+    if (this.disabled) {
       event.preventDefault();
       event.stopPropagation();
-    } else {
-      this.focus();
+    }
+
+    return this.disabled;
+  }
+}
+
+
+/**
+ * Applies proper (click) support and adds styling for use with the Material Design "cancel" icon
+ * available at https://material.io/icons/#ic_cancel.
+ *
+ * Example:
+ *
+ *     <md-chip>
+ *       <md-icon mdChipRemove>clear</md-icon>
+ *     </md-chip>
+ *
+ * You *may* use a custom icon, but you may need to override the `md-chip-remove` positioning styles
+ * to properly center the icon within the chip.
+ */
+@Directive({
+  selector: '[mdChipRemove], [matChipRemove]',
+  host: {
+    'class': 'mat-chip-remove',
+    '[class.mat-chip-remove-hidden]': '!visible',
+    '(click)': '_handleClick($event)'
+  }
+})
+export class MdChipRemove {
+
+  /** Whether or not the remove icon is visible. */
+  _isVisible: boolean = true;
+
+  @Input('mdChipRemoveVisible')
+  get visible() { return this._isVisible; }
+  set visible(value) { this._isVisible = coerceBooleanProperty(value); }
```

Comment:
Why does this have a `visible` property built-in? Wouldn't someone just hide it like anything else (css or `ngIf`)?